src/android/DecryptionExecutor.java

```diff
@@ -0,0 +1,32 @@
+package com.crypho.plugins;
+
+
+import android.util.Log;
+import java.util.concurrent.Callable;
+
+
+public class DecryptionExecutor implements Callable<String> {
+    private static final String TAG = "SecureStorage";
+    private String alias;
+    private byte[] encKey, iv, ct, adata;
+
+    public DecryptionExecutor(byte[] encKey, String alias, byte[] iv, byte[] ct, byte[] adata) {
+        this.encKey = encKey;
+        this.alias = alias;
+        this.iv = iv;
+        this.ct = ct;
+        this.adata = adata;
+    }
+
+    @Override
+    public String call() {
+        try {
+            byte[] decryptedKey = RSA.decrypt(encKey, alias);
+            String decrypted = new String(AES.decrypt(ct, decryptedKey, iv, adata));
+            return decrypted;
+        } catch (Exception e) {
+            Log.e(TAG, "Decrypt (RSA/AES) failed :", e);
+            return e.getMessage();
+        }
+    }
```

Comment:
Maybe is a technical decision. But why not throw the error? Seems like if it does error on decryption you need to check the String for an error message.

Reply:
Simply maintaining the existing login, didn't add or subtract anything, merely moved it's place

Reply:
This isn't maintaining the existing logic. The existing logic calls either success or error in the callbackContext depending on an exception being thrown or not. That's not what happens here, all is being converted into a string. The the contains "ERROR" is used, which is incorrect, for 2 reasons:
- A successful value can very well contain "ERROR", since it's defined by the plugin user
- We can't be sure these error messages will contain "ERROR" in future SDKs

We need to return an object that includes both a success boolean and the result. Create a POJO for it or reuse the JSONObject.

Reply:
Added POJO